pkg/controllers/manifestwork.go

```diff
@@ -0,0 +1,250 @@
+		if err := r.Get(ctx, types.NamespacedName{Namespace: getTargetNamespace(hyd), Name: np.Name}, &nodePool); err == nil {
+			r.Log.Info(fmt.Sprintf("Waiting for NodePool %s/%s to be deleted", getTargetNamespace(hyd), np.Name))
+			return ctrl.Result{RequeueAfter: 10 * time.Second, Requeue: true}, nil
+		} else {
+			r.Log.Info(fmt.Sprintf("NodePool %s/%s already deleted...", getTargetNamespace(hyd), hyd.Name))
+		}
+	}
+
+	// Delete the HostedCluster
+	var hc hyp.HostedCluster
+	if err := r.Get(ctx, types.NamespacedName{Namespace: getTargetNamespace(hyd), Name: hyd.Name}, &hc); !apierrors.IsNotFound(err) {
+		r.Log.Info(fmt.Sprintf("Waiting for HostedCluster %s/%s to be deleted", getTargetNamespace(hyd), hyd.Name))
+		return ctrl.Result{RequeueAfter: 10 * time.Second, Requeue: true}, nil
+	} else {
+		r.Log.Info(fmt.Sprintf("HostedCluster %s/%s already deleted...", getTargetNamespace(hyd), hyd.Name))
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *HypershiftDeploymentReconciler) appendReferenceSecrets(ctx context.Context, hyd *hypdeployment.HypershiftDeployment) (loadManifest, error) {
+	cpoCreds := &corev1.Secret{}
+	if err := r.Get(ctx, types.NamespacedName{Name: hyd.Spec.HostedClusterSpec.Platform.AWS.ControlPlaneOperatorCreds.Name,
+		Namespace: hyd.GetNamespace()}, cpoCreds); err != nil {
+		return nil, fmt.Errorf("failed to get the cpo creds, err: %w", err)
+	}
+
+	kccCreds := &corev1.Secret{}
+	if err := r.Get(ctx, types.NamespacedName{Name: hyd.Spec.HostedClusterSpec.Platform.AWS.KubeCloudControllerCreds.Name,
+		Namespace: hyd.GetNamespace()}, kccCreds); err != nil {
+		return nil, fmt.Errorf("failed to get the cpo creds, err: %w", err)
+	}
+
+	nmcCreds := &corev1.Secret{}
+	if err := r.Get(ctx, types.NamespacedName{Name: hyd.Spec.HostedClusterSpec.Platform.AWS.NodePoolManagementCreds.Name,
+		Namespace: hyd.GetNamespace()}, nmcCreds); err != nil {
+		return nil, fmt.Errorf("failed to get the cpo creds, err: %w", err)
+	}
+
+	pullCreds := &corev1.Secret{}
+	if err := r.Get(ctx, types.NamespacedName{Name: hyd.Spec.HostedClusterSpec.PullSecret.Name,
+		Namespace: hyd.GetNamespace()}, pullCreds); err != nil {
+		return nil, fmt.Errorf("failed to get the cpo creds, err: %w", err)
+	}
+
+	tempSecret := func(in *corev1.Secret) *corev1.Secret {
+		out := &corev1.Secret{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "Secret",
+				APIVersion: corev1.SchemeGroupVersion.String(),
+			},
+		}
+
+		out.SetName(in.GetName())
+		out.SetNamespace(getTargetNamespace(hyd))
+		out.SetLabels(in.GetLabels())
+		out.Data = in.Data
+
+		return out
+	}
+
+	refSecrets := []*corev1.Secret{cpoCreds, kccCreds, nmcCreds, pullCreds}
+
+	return func(hyd *hypdeployment.HypershiftDeployment, payload *[]workv1.Manifest) {
+		for _, s := range refSecrets {
+			o := tempSecret(s)
+			*payload = append(*payload, workv1.Manifest{RawExtension: runtime.RawExtension{Object: o}})
+		}
+
+	}, nil
+}
+
+func getTargetNamespace(hyd *hypdeployment.HypershiftDeployment) string {
```

Comment:
Is this where the ToDo will be to make sure that a clusterSet is bound to this namespace, and that the target ManagedCluster is a member of the clusterSet?

Reply:
yup, I will put a todo in comment